flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/sink/committables/SubtaskCommittableManager.java

```diff
@@ -72,7 +86,7 @@ void add(CommittableWithLineage<CommT> committable) {
 
     void add(CommT committable) {
         checkState(requests.size() < numExpectedCommittables, "Already received all committables.");
-        requests.add(new CommitRequestImpl<>(committable));
+        requests.add(new CommitRequestImpl<>(committable, metricGroup));
```

Comment:
Increment total # of committables here instead of within `CommitRequestImpl` constructor.

Reply:
Moved